File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        // int [] res=new int[intervals.length];
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        minHeap.add(intervals[0][1]);
        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] >= minHeap.peek()) {
                minHeap.poll();
            }
            minHeap.add(intervals[i][1]);
        }
        return minHeap.size();
    }
}

File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.py
import heapq
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        if len(intervals) == 0 :
            return -1
        intervals.sort(key=lambda x:x[0])
        minHeap = []
        for start,end in intervals:
            if minHeap and start >= minHeap[0]:
                heapq.heapreplace(minHeap,end)
            else:
                heapq.heappush(minHeap,end)
        return len(minHeap)
        